File: services/api/src/atlas_api/models/job.py
"""Job domain models — mirrors packages/shared/src/types/job.ts."""

from typing import Any

from pydantic import BaseModel, ConfigDict

from atlas_api.models.enums import JobKind, JobStatus


class JobError(BaseModel):
    model_config = ConfigDict(frozen=True)

    code: str
    message: str
    retryable: bool


class Job(BaseModel):
    model_config = ConfigDict(frozen=True)

    id: str
    workspace_id: str
    kind: JobKind
    status: JobStatus
    payload: dict[str, Any]
    result: dict[str, Any] | None
    error: JobError | None
    attempt: int
    max_attempts: int
    queued_at: str
    started_at: str | None
    completed_at: str | None

File: packages/shared/src/types/job.ts
import type { JobId, WorkspaceId } from "./ids.js";
import type { JobKind, JobStatus } from "./enums.js";

export interface JobError {
  readonly code: string;
  readonly message: string;
  readonly retryable: boolean;
}

export interface Job {
  readonly id: JobId;
  readonly workspaceId: WorkspaceId;
  readonly kind: JobKind;
  readonly status: JobStatus;
  readonly payload: Readonly<Record<string, unknown>>;
  readonly result: Readonly<Record<string, unknown>> | null;
  readonly error: JobError | null;
  readonly attempt: number;
  readonly maxAttempts: number;
  readonly queuedAt: string;
  readonly startedAt: string | null;
  readonly completedAt: string | null;
}
